src/main/java/fi/hsl/common/monitoring/MonitoringScheduler.java

```
@@ -0,0 +1,81 @@
+
+    public static MonitoringScheduler getInstance(final PulsarApplicationContext context, final long delay, final long interval, final Runnable runnable) {
+        producer = context.getProducer();
+        MonitoringScheduler.delay = delay;
+        MonitoringScheduler.interval = interval;
+        MonitoringScheduler.runnable = runnable;
+        return ourInstance;
+    }
+
+    public static MonitoringScheduler getInstance(final PulsarApplicationContext context, final long delay, final long interval) {
+        return getInstance(context, delay, interval, ourInstance.getRunnable());
+    }
+
+    protected MonitoringScheduler() {}
+
+    public boolean addHandler(final AbstractMonitoringHandler handler) {
+        if (!handlers.contains(handler)) {
+            return handlers.add(handler);
+        }
+        return false;
+    }
+
+    public boolean removeHandler(final AbstractMonitoringHandler handler) {
+        return handlers.remove(handler);
+    }
+
+    protected Runnable getRunnable() {
+        return () -> {
+            final long timestamp = System.currentTimeMillis();
+            handlers.stream().forEach(handler -> {
+                final String key = handler.getKey();
+                final double value = handler.getValueAndClearSync();
+                Monitoring.MonitoringMessage message = MonitoringMessageFactory.createMonitoringMessage(timestamp, key, value);
+                TypedMessageBuilder<byte[]> messageBuilder = producer.newMessage()
+                        .eventTime(timestamp)
+                        .value(message.toByteArray());
+                messageBuilder.sendAsync()
+                        .whenComplete((id, throwable) -> {
+                            if (throwable != null) {
+                                LOGGER.warn("Failed to send Pulsar message.", throwable);
+                            }
+                        });
+            });
```

Java 8 stream().forEach will swallow all exceptions (It's a real pain in the ass..). Would be nice to do something on errors, at least log them?

---

shiit, good to know.